src/workerd/jsg/resource.h

```diff
@@ -966,6 +1086,28 @@ struct ResourceTypeBuilder {
             v8::Local<v8::Value>(), signature, 0, v8::ConstructorBehavior::kThrow));
   }
 
+  // Register a method with both slow and fast paths using the V8 Fast API mechanism
+  template <const char* name, typename Method, Method method, bool trackCallCount = false>
```

Comment:
Perhaps instead of passing `trackCallCount` as a template option and having the additional jsg macro, can enabling the call count tracking be automatically enabled only when running in predicable mode for tests?

Reply:
`isPredictableModeForTest` is not a constexpr method, this means that if we use it, registerFastMethod on production will always pay a toll for an unnecessary branch